TOOLS_OVERVIEW.md

```diff
@@ -0,0 +1,228 @@
+has an overview of `goto-instrument` and a small tutorial example.
+- [Developer Tutorial](http://cprover.diffblue.com/tutorial.html) has high
+level overview and some example commands for `goto-instrument`.
+
+
+## janalyzer
+
+This provides a way to access and invoke various forms of analysis on
+Java programs.
+
+Documentation useful for this tool can be found:
+- [Analysis Information](http://cprover.diffblue.com/group__analyses.html)
+
+Details of all the options can be seen by running
+```
+janalyzer --help
+```
+
+## java-unit
+
+Runs Java unit tests. For more details use
+```
+java-unit --help
+```
+Default behaviour is to only show failed test cases.
+
+
+## jbmc
+
+This is the main analysis engine the performs the analysis
+of Java files using bounded model checking. This is a
+Java version of the `cbmc` tool (more documentation for
+`cbmc` is available and much of it applies to `jbmc`
+since they use many of the same back end utilities).
+
+For some light information on usage of the jbmc tool see the following documentation
+- [JBMC homepage](https://www.cprover.org/jbmc/) includes some
+simple examples and information.
+
+For details on how analysis is performed in the `cbmc` and
+`jbmc` tools see see the following documents:
+- [CBMC Architecture](http://cprover.diffblue.com/cbmc-architecture.html)
+gives a high level overview of the `cbmc` architecture and data flow that *should also apply to* `jbmc`.
+- [Background Concepts](http://cprover.diffblue.com/background-concepts.html)
+overviews all the key concepts used in the `jbmc` analysis.
+
+For details on compiling, testing, contributing, and documentation related to development see:
+- [CProver Development Documentation](http://cprover.diffblue.com/index.html)
+
+
+## jdiff
+
+Provides a variety of difference checks between two goto programs (produced
+by `goto-cc`). This invokes some of the `cbmc` tools to convert the goto
+program and then determine which functions are added/removed/changed.
+
+Details of all the options can be seen by running
+```
+jdiff --help
+```
+this includes both options for the difference, and options for the goto
+program instrumentation.
+
+
+## memory-analyzer
+
+This is a wrapper program that provides a front end to `gdb` that adds some
+useful features related to the other goto utilities. Note that to use
+`memory-analyzer` the program must be compiled with `goto-cc`. To make
+best use of `memory-analyzer` and `gdb` you should compile with the `-g`
+option. The `memory-analyzer` can also be used to create harnesses for use
+with `goto-harness`.
+
+For further documentation and examples see 
+[here](http://cprover.diffblue.com/md__home_travis_build_diffblue_cbmc_doc_architectural_memory-analyzer.html).
+
+
+## smt2_solver
+
+This is an (Satisfiability Modulo Theory) SMT solver that is shipped
+with CProver. The SMT solver here is built on top of a Satisfiability
+(SAT) solver. The `smt2_solver` is designer to work with files, and
```

Comment:
It is basically an SMT interface to the CPROVER back-end prover.

Reply:
@martin-cs Can you please clarify whether you are proposing a change to the text (and if so perhaps propose some new text), or whether you were making a comment to help us understand?

Reply:
I think more of the later but let's try the former too...

`smt2_solver` is an SMT (Satisfiability Modulo Theory) solver.  It parses SMT-LIB 2 format and uses CPROVER's internal bit-blasting solver (see `solvers/`) to resolve queries.